Merge flappy and flappy_birds

Dimi20cen opened the pull request:
Merge flappy with flappy-birds

- Add up-arrow key functionality for bird movement
- Refine bird rotation for smoother gameplay
- Implement a moving background for enhanced visuals
- Incorporate flapping wings animation
- Adjust the starting point of bottom pipes for better visuals
- Enhance collision detection, using masks, for increased precision
- Introduce a sound effect upon collision, with the bird falling to the ground
- Tweak the points sound effect timing for improved feedback
- Update the game font and add an outline, to resemble the original Flappy Bird
- Replace the existing Sugargame file with the latest version
- Add flake8 file

@chimosky @sourabhaa

Dimi20cen committed:
Add movement to background

Dimi20cen committed:
Made floor speed adjust to difficulty

Dimi20cen committed:
Aesthetic changes to pipes

- Pipes now "sit" on top of the floor
- Score is now incremented after surpassing the pipe

Dimi20cen committed:
Change text style

Replaced font to match the game character better

Dimi20cen committed:
Change bird movement and visuals

Dimi20cen committed:
Make collisions more accurate

Dimi20cen committed:
Update sugargame dir and add .flake8

chimosky commented on bird.py:
```diff
@@ -3,52 +3,67 @@
+        self.index = 0
+        self.counter = 0
+        self.count_flap = 0
+        for num in range(0, 3):
```
You don't need to specify 0 here.

Dimi20cen replied:
Do you suggest I shouldn't initialize index, counter and count_flap?

chimosky replied:
Not at all, I'm saying your range can be `range(3)` as you're not using index, counter or count_flap in your range loop.

chimosky commented on main.py:
```diff
@@ -211,9 +217,14 @@ def run(self):
 
 if __name__ == "__main__":
     g = Flappy()
+    # GAME_SIZE = (1920, 900)
+    # GAME_SIZE = (1024, 768)
+    # GAME_SIZE = (684, 600)
```
You can delete these.

Dimi20cen replied:
I left them in for more convient testing outside of sugar env, is it not good practice?

chimosky replied:
As they're for testing, they shouldn't be included in your commit as you're done with testing and have settled on what to use.

chimosky commented:
@Dimi20cen I opened a PR in your master branch, couldn't push here as you're working on the master branch.

Dimi20cen committed:
Simplify range usage and address reviewer feedback

Changed range(0, 3) to range(3) for clarity

Dimi20cen committed:
Adress reviewer feedback delete test values

sourabhaa commented:
tested, I'm able to see the toolbar,
![Screenshot of _Flappy Activity_](https://github.com/sugarlabs/flappy/assets/43743186/88c5fda3-0b59-456d-8aa5-3591ea567d94)

jriyyya commented:
Tested [dec589d](https://github.com/sugarlabs/flappy/pull/18/commits/dec589d756c025c25a74736665cd20b79039311f), Toolbar is visible and functional to me

chimosky commented:
Tested dec589d and the toolbar shows, I'll compare with flappy birds before merging.

chimosky commented:
I compared with flappy birds activity and they're a few things left;

- [ ] sound; Flappy birds have sound from start of the game till the end of the game.
- [ ] Game speed; flappy seems to have a very slow game speed compared to flappy birds and it would be great to increase the game speed as it's more interesting.

Dimi20cen commented:
> sound; Flappy birds have sound from start of the game till the end of the game.

Could you clarify? I think the merged activity and flappy-birds have, identical sounds.

> Game speed; flappy seems to have a very slow game speed compared to flappy birds and it would be great to increase the game speed as it's more interesting.

While x-speed is slower than flappy's, I aimed to mimic the x-speed of the original implementation of the game flappy birds, trusting that they'd found the sweet spot.

chimosky commented:
> Could you clarify? I think the merged activity and flappy-birds have, identical sounds.

There are no sounds when I test your changes, @jriyyya @sourabhaa was the sound working fine when you tested?

> While x-speed is slower than flappy's, I aimed to mimic the x-speed of the original implementation of the game flappy birds, trusting that they'd found the sweet spot.

Okay.

Dimi20cen committed:
Add sound for hit

Previous to this addition, sounds would not be working, since the try-except condition in line 154 to 163 would go to except because hit.ogg was missing.

With the addition of hit.ogg in data/sounds, sound issue should be fixed.

Dimi20cen commented:
Sound should be fixed @chimosky.

chimosky commented:
Tested c6062e1, sound now works as expected. Thanks!
